### PR TITLE
GCE storage profiles: FLOC-3828

### PR DESCRIPTION
**Pull request by justnoise**
Not much to say about this one.  GCE only has 2 types of disks we can use: magnetic PD and SSD PD so bronze will be magnetic, gold and silver will be SSD.  The hoops to use Values and ValueConstants make me a bit sad but work...  Feel free to suggest alternatives.

### CODE REVIEW

File: flocker/node/agents/functional/test_gce.py [hunk 1]
```diff
@@ -24,23 +24,21 @@
 """
 
 from uuid import uuid4
+from bitmath import GiB
```
Comment (wallrj) [inline flocker/node/agents/functional/test_gce.py]: Unused import

File: flocker/node/agents/functional/test_gce.py [hunk 2]
```diff
@@ -24,23 +24,21 @@
...
-from ..blockdevice import AlreadyAttachedVolume
-
-from ..gce import get_machine_zone, get_machine_project
+from ..blockdevice import AlreadyAttachedVolume, MandatoryProfiles
+from ..gce import get_machine_zone, get_machine_project, GCEDiskTypes
 from ....provision._gce import GCEInstanceBuilder
-
 from ..test.test_blockdevice import (
-    make_iblockdeviceapi_tests
+    make_iblockdeviceapi_tests, make_iprofiledblockdeviceapi_tests,
```
Comment (wallrj) [inline flocker/node/agents/functional/test_gce.py]: How come we don't use these standard tests with GCE?

Comment (wallrj) [inline flocker/node/agents/functional/test_gce.py]: Maybe remove this import and add an explanatory note if those tests can't be used for some reason.

File: flocker/node/agents/functional/test_gce.py [hunk 3]
```diff
@@ -113,6 +111,40 @@ def test_attach_elsewhere_attached_volume(self):
         # the hack.
         pass
 
+    def test_create_volume_gold_profile(self):
```
Comment (wallrj) [inline flocker/node/agents/functional/test_gce.py]: Add some short docstrings for these new tests. 
Or replace them with `make_iprofile_blockdevice_tests` if possible.

File: flocker/node/agents/gce.py [hunk 4]
```diff
@@ -411,6 +426,12 @@ def list_volumes(self):
                 disk['description'] == self._disk_resource_description())
         )
 
+    def _get_gce_volume(self, blockdevice_id):
```
Comment (wallrj) [inline flocker/node/agents/gce.py]: Add docstring.

**Comment by wallrj** [comment]
Thanks @justnoise 

Looks good to me. I made a few comments. Please answer or address those and merge when the tests are green.
Oh, and it looks like @sarum90 landed another conflicting branch, so you'll have to merge forward.

**Comment by justnoise** [comment]
I've moved the storage profile tests into `GCEProfiledBlockDeviceApiTests` which is much better.

**Pull request merged**